src/main/java/uk/gov/hmcts/reform/ucmc/service/PaymentsService.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.ucmc.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.payments.client.PaymentsClient;
+import uk.gov.hmcts.reform.payments.client.models.FeeDto;
+import uk.gov.hmcts.reform.payments.client.models.PaymentDto;
+import uk.gov.hmcts.reform.payments.client.request.CreditAccountPaymentRequest;
+import uk.gov.hmcts.reform.ucmc.config.PaymentsConfiguration;
+import uk.gov.hmcts.reform.ucmc.request.RequestData;
+
+import java.math.BigDecimal;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class PaymentsService {
+
+    private final FeesService feesService;
+    private final PaymentsClient paymentsClient;
+    private final RequestData requestData;
+    private final PaymentsConfiguration paymentsConfiguration;
+
+    public PaymentDto createCreditAccountPayment(CaseDetails caseDetails) {
+        var claimValue = new BigDecimal(caseDetails.getData().get("claimValue").toString());
```

Comment:
ClaimValue follows the structure claimValue.higherValue now. There should be an object for it merged into master with 328

Reply:
updated